Posit Infrastructure/Guide to using renv to create reproducible projects.md

```diff
@@ -0,0 +1,47 @@
+
+## Step-by-step instructions
+
+Pre-requisite: a project (preferably version-controlled using git).
+
+#### Setting up [{renv}](https://rstudio.github.io/renv/) for the first time in a project:
+1.	Open the project.
+2.	Call `renv::init()` to initialise renv within that project. This can take a while depending on the size of the project.
+3.	Calling `renv::status()` should confirm the new 'lockfile' is synced with the project (i.e., contains details of all the required packages).
+4.	Commit the changes (additional files and folders) to version control.
+
+#### Adding (/removing) packages to the project:
+1.	Install (or remove) the required packages as usual. 
+2.	Change the code, by adding (or removing) calls to `library()` or `require()`, or the package function calls themselves e.g. `dplyr::group_by()`.
+3.	Calling `renv::status()` will tell you there are changes that aren’t recorded in the lockfile.
+4.	Calling `renv::snapshot()` will record any changes into the lockfile.
+5.	Calling `renv::status()` should now confirm everything's synced. 
+6.	Commit the changes in the renv.lock file using version control.
+
+#### Using a project that has already had [{renv}](https://rstudio.github.io/renv/) initialised (e.g., when collaborating on a colleague's project):
+1.	Ensure the project is synced with the latest version-controlled repository (e.g., `git pull` from github).
+2.	Call `renv::restore()` to make sure your local version of the project has all the required packages.
+3.	Calling `renv::status()` should now confirm everything's synced. 
+
+## More detail about what [{renv}](https://rstudio.github.io/renv/) does
+
+[{renv}](https://rstudio.github.io/renv/) works by storing the exact versions of each package used within each project. This helps to isolate the project from any external changes that might have caused problems (e.g., updating/removing packages when working on a different project). Once [{renv}](https://rstudio.github.io/renv/) is initialised in a project it stays on unless deliberately turned off (see [Introduction to renv](https://rstudio.github.io/renv/articles/renv.html)). 
+
+When you initialise [{renv}](https://rstudio.github.io/renv/) in a project (renv::init()) it searches the scripts for any calls to `library()`, `require()` or package function calls and identifies which versions of each package are in use from the versions installed in the user's package library.. The following files/folders are then added to the project:
+* The folder `renv/library`: this is used for storing the packages. If the project is version-controlled using git a `.gitignore` file will be written to `renv/`, so that this library is ignored when changes to the project are committed.  
+* The lockfile `renv.lock`: this stores the details of the packages (including the versions used). 
+* The profile file `.Rprofile`: this is run automatically when the project is opened, to ensure it uses the packages stored in renv/library.
```

Comment:
What would happen is someone is adding renv capability to a project that already has an .Rprofile in place? Would a new one still be created or would the pre-existing one just be updated?

Reply:
The `.Rprofile` is just updated in this case. It's one line with a call to `renv/activate.R`. I'm not aware of any conflicts or issues that would occur here as it's common to use the `.Rprofile` for things like environment variables, I assume it's well-tested.